### PR TITLE
Default value for ModelCheckpoint filepath

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1535.
As the issue outlines, it is not easy for the user to change some of the argument in `ModelCheckpoint` (like `save_top_k`) and still keep the default filepath because it is determined at runtime. This PR implements the option to pass in `filepath=None` which will be detected at runtime, and then filepath will be set to the default location. 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by ematvey** [comment]
Just an idea, it would be even better to allow user to specify save path and checkpoint filename template separately. I might want to override the template also to include useful metrics, etc, but I still wouldn't want to mess with default save path.

**Comment by SkafteNicki** [comment]
I agree that it would be nice to split `filepath` into two arguments, a `dirpath` and a `filename`. I do not think it is possible to do with complete backward compatibility. Such a change would need confirmation from the core contributors.

**Comment by ematvey** [comment]
We could add `savedir` and `filename` (None defaults) and use them when `filename` is defined, otherwise fallback to `filepath`, then gradually deprecate `filepath`.

**Comment by SkafteNicki** [comment]
Nice idea, let me try to implement it. From what future version should `filepath` be deprecated?

**Comment by Borda** [comment]
> We could add `savedir` and `filename` (None defaults) and use them when `filename` is defined, otherwise fallback to `filepath`, then gradually deprecate `filepath`.

> Nice idea, let me try to implement it. From what future version should `filepath` be deprecated?

sounds good to me...
cc: @PyTorchLightning/core-contributors

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -90,12 +94,14 @@ def __init__(self, filepath: str, monitor: str = 'val_loss', verbose: bool = Fal
 
         self.monitor = monitor
         self.verbose = verbose
-        if os.path.isdir(filepath):
-            self.dirpath, self.filename = filepath, '{epoch}'
+        if filepath is None:  # will be determined by trainer at runtime
```
Comment (Borda) [inline pytorch_lightning/callbacks/model_checkpoint.py:97]: it is will be deprecated there shall be warning...

Comment (Borda) [reply pytorch_lightning/callbacks/model_checkpoint.py:97]: @SkafteNicki mind add deprecation warning?

**Comment by SkafteNicki** [comment]
@Borda the changes have already been merged, before I had time to do the change we discussed. Should I go ahead and make a new PR?

**Comment by Borda** [comment]
> @Borda the changes have already been merged, before I had time to do the change we discussed. Should I go ahead and make a new PR?

Yes please

**Pull request merged**